dmcontent/markdown.py

```diff
@@ -0,0 +1,42 @@
+from typing import Mapping
+from xml.etree.ElementTree import Element
+
+from markdown.extensions import Extension
+from markdown.treeprocessors import Treeprocessor
+
+
+class _ClassAddingTreeprocessor(Treeprocessor):
+    _mapping: Mapping[str, str]
+
+    def __init__(self, mapping: Mapping[str, str]):
+        self._mapping = mapping
+
+    def run(self, root: Element):
+        self._run(root)
+
+    def _run(self, element: Element):
+        # recurse
+        for child in element:
+            self._run(child)
+
+        # mutate each affected element in-place
+        if element.tag in self._mapping:
+            orig_class = element.attrib.get("class", "")
+            element.set("class", (orig_class + " " if orig_class else "") + self._mapping[element.tag])
+
+
+class GovUkFrontendExtension(Extension):
+    """
+    Markdown extension adding govuk-frontend classes to generated paragraphs and lists (note - will not affect
+    manual html)
+    """
+    def extendMarkdown(self, md, md_globals):
+        # NOTE the interface for registering extensions is much improved after Markdown 3.0.0. TODO upgrade.
+        md.registerExtension(self)
+        self.processor = _ClassAddingTreeprocessor({
+            "p": "govuk-body",
+            "ul": "govuk-list govuk-list--bullet",
+            "ol": "govuk-list govuk-list--number",
```

Comment:
Could we pull this out separately so it's super easy to append additional classes to later if needed?

Reply:
Yes, there's even a mechanism for doing that built into the extension business but I decided not to wrestle with that now seeing as I didn't even know where the config should be kept.

Reply:
Done